### PR TITLE
AK: Add `nodiscard` attribute

### PR DESCRIPTION
**Pull request by ldm5180**
- AllOf
- AnyOf
- Base64
- BinaryHeap
- BinarySearch
- BitCast
- BitStream
- Find

### CODE REVIEW

File: AK/Base64.h [hunk 1]
```diff
@@ -13,14 +13,13 @@
...
-size_t calculate_base64_decoded_length(StringView);
+[[nodiscard]] size_t calculate_base64_decoded_length(StringView);
 
-size_t calculate_base64_encoded_length(ReadonlyBytes);
+[[nodiscard]] size_t calculate_base64_encoded_length(ReadonlyBytes);
 
-ErrorOr<ByteBuffer> decode_base64(StringView);
-
-String encode_base64(ReadonlyBytes);
+[[nodiscard]] ErrorOr<ByteBuffer> decode_base64(StringView);
```
Comment (linusg) [inline AK/Base64.h:20]: `ErrorOr` is already `nodiscard` - I'm not against stating it explicitly here again, but then I'm not sure what the point of marking classes themselves as `nodiscard` is :)

**Review by linusg — COMMENTED** [review]
Out of curiosity, is this all of AK where `nodiscard` would be useful, or just a (random?) subset?

**Comment by ldm5180** [comment]
> Out of curiosity, is this all of AK where `nodiscard` would be useful, or just a (random?) subset?

@linusg It is just the ones I have had time to do so far. I can do them all before making the PR if that is preferred.

File: AK/Base64.h (hunk 1 above)
Comment (ldm5180) [reply AK/Base64.h:20]: That is a better way of doing it. I will remove it from these spots.

**Pull request merged**